website/docs/concepts/method.mdx

```diff
@@ -131,4 +131,107 @@ const endpoint = createEndpoint({
 export default endpoint.handler;
 ```
 
-<!-- choosing not to document parsing/validation yet due to #3 -->
+### Parsing
+
+To verify and optionally transform parts of the original request, you can provide parsers. These will receive the original data from a request, and are expected to return data that will be passed to the final handler, or throw an error.
+
+Currently, you can provide parsers for `body` and `query`.
+
+```ts title="pages/api/foo.ts"
+// file: src/api/index.ts noEmit
+import { createEndpointFactory } from 'next-create-endpoint-factory';
+
+export const createEndpoint = createEndpointFactory();
+
+// file: pages/api/foo.ts
+
+import { createEndpoint } from '../../src/api';
+
+const endpoint = createEndpoint({
+  methods: (method) => ({
+    post: method<{ created: true }>()({
+      parsers: {
+        body: (body, failWithCode) => {
+          if (!body) {
+            throw failWithCode(400, 'no body provided');
+          }
+          if (typeof body !== 'string') {
+            throw failWithCode(400, 'invalid body');
+          }
+          return body; // now string
+        },
+      },
+      handler: ({ body }, { failWithCode, succeedWithCode }) => {
+        console.log(body); // typed as string thanks to parser
+        return succeedWithCode(201, { created: true });
+      },
+    }),
+  }),
+});
+
+export default endpoint.handler;
+```
+
+:::caution
+When using Typescript, it's required to use the double call syntax `method<ReturnType>()(definition)` to use parsers.
+
+This allows the `ReturnType` type to be provided explicitly, while inferring the parsed types.
+
+If you're using Javascript, this is not required.
+:::
+
+<!-- TODO: work out why the below recipe results in body being typed as `any`
```

Comment:
worth creating an issue for this if you haven't already?

Reply:
https://github.com/actual-experience/create-endpoint-factory/issues/6